src/test/java/de/tum/cit/aet/artemis/core/security/lti/Lti13LaunchFilterTest.java

```diff
@@ -104,20 +107,37 @@ void init() {
         launchFilter = new Lti13LaunchFilter(defaultFilter, CustomLti13Configurer.LTI13_LOGIN_PATH, lti13Service);
         SecurityContextHolder.setContext(securityContext);
         doReturn(authentication).when(securityContext).getAuthentication();
-        doReturn(CustomLti13Configurer.LTI13_LOGIN_PATH).when(httpRequest).getServletPath();
 
+        // Initialize a real MockHttpServletRequest
+        httpRequest = new MockHttpServletRequest();
+
+        // Set servlet path (this was previously mocked via getServletPath())
+        httpRequest.setServletPath(CustomLti13Configurer.LTI13_LOGIN_PATH);
+
+        // Set attributes (previously mocked getAttribute)
+        HttpServletMapping mapping = mock(HttpServletMapping.class);
+        doReturn(MappingMatch.EXACT).when(mapping).getMappingMatch();
+        httpRequest.setAttribute(RequestDispatcher.INCLUDE_MAPPING, mapping);
+
+        // Simulate the INCLUDE_REQUEST_URI_ATTRIBUTE used by Spring (was mocked)
+        httpRequest.setAttribute(WebUtils.INCLUDE_REQUEST_URI_ATTRIBUTE, CustomLti13Configurer.LTI13_LOGIN_PATH);
```

Comment:
For me it would have been easier to understand if the block was grouped together and not spread apart by the comments - I would rather have extended the initial comment `Initialize a real MockHttpServletRequest` with a rationale why we made that decision

I do also think that "was mocked" will be rather confusing once this PR is merged
```suggestion
        // Initialize a real MockHttpServletRequest
        httpRequest = new MockHttpServletRequest();
        httpRequest.setServletPath(CustomLti13Configurer.LTI13_LOGIN_PATH);
        HttpServletMapping mapping = mock(HttpServletMapping.class);
        doReturn(MappingMatch.EXACT).when(mapping).getMappingMatch();
        httpRequest.setAttribute(RequestDispatcher.INCLUDE_MAPPING, mapping);
        httpRequest.setAttribute(WebUtils.INCLUDE_REQUEST_URI_ATTRIBUTE, CustomLti13Configurer.LTI13_LOGIN_PATH);
```